packages/delegation-toolkit/src/actions/getCaveatAvailableAmount.ts

```diff
@@ -0,0 +1,286 @@
+
+/**
+ * Parameters for ERC20 streaming enforcer.
+ */
+export type ERC20StreamingParams = BaseCaveatParams;
+
+/**
+ * Parameters for MultiTokenPeriodEnforcer.
+ */
+export type MultiTokenPeriodParams = {
+  args: Hex;
+} & BaseCaveatParams;
+
+/**
+ * Parameters for native token period transfer enforcer.
+ */
+export type NativeTokenPeriodTransferParams = BaseCaveatParams;
+
+/**
+ * Parameters for native token streaming enforcer.
+ */
+export type NativeTokenStreamingParams = BaseCaveatParams;
+
+/**
+ * Return type for period-based transfer enforcers
+ */
+export type PeriodTransferResult = {
+  availableAmount: bigint;
+  isNewPeriod: boolean;
+  currentPeriod: bigint;
+};
+
+/**
+ * Return type for streaming enforcers
+ */
+export type StreamingResult = {
+  availableAmount: bigint;
+};
+
+/**
+ * Resolves the delegation manager address from parameters or environment.
+ *
+ * @param params - The parameters object.
+ * @param params.delegationManager - The delegation manager address.
+ * @param environment - The delegator environment.
+ * @returns The resolved delegation manager address.
+ */
+function resolveDelegationManager(
+  params: { delegationManager?: Address },
+  environment: DeleGatorEnvironment,
+): Address {
+  const delegationManager =
+    params.delegationManager ?? environment.DelegationManager;
+
+  if (!delegationManager) {
+    throw new Error('Delegation manager address not found');
+  }
+
+  return delegationManager;
+}
+
+/**
+ * Resolves the enforcer address from parameters or environment.
+ *
+ * @param enforcerName - The name of the enforcer.
+ * @param params - The parameters object.
+ * @param params.enforcerAddress - The enforcer address.
+ * @param environment - The delegator environment.
+ * @returns The resolved enforcer address.
+ */
+function resolveEnforcerAddress(
+  enforcerName: keyof DeleGatorEnvironment['caveatEnforcers'],
+  params: { enforcerAddress?: Address },
+  environment: DeleGatorEnvironment,
+): Address {
+  if (params.enforcerAddress) {
+    return params.enforcerAddress;
+  }
+
+  const enforcerAddress = environment.caveatEnforcers[enforcerName];
+  if (!enforcerAddress) {
+    throw new Error(`${enforcerName} not found in environment`);
+  }
+
+  return enforcerAddress;
+}
+
+/**
+ * Caveat enforcer actions for extending viem clients.
+ *
+ * @param config - The configuration for caveat enforcers.
+ * @returns The client extension with caveat enforcer actions.
+ */
+export const caveatEnforcerActions =
+  (config: CaveatEnforcerClientConfig) => (client: PublicClient) => ({
+    /**
+     * Get available amount for ERC20 period transfer enforcer.
+     *
+     * @param params - The parameters for the ERC20 period transfer enforcer.
+     * @returns Promise resolving to the period transfer result.
+     */
+    getErc20PeriodTransferEnforcerAvailableAmount: async (
+      params: ERC20PeriodTransferParams,
+    ): Promise<PeriodTransferResult> => {
```

Comment:
It might be nice to export these actions individually too, and from @metamask/delegation-toolkit/actions

then people can be all:
```
const client = createPublicClient({...});

getErc20PeriodTransferEnforcerAvailableAmount(
  client,
  {
    ...
});
```

note: the action interface is `(client, config)` rather than `({ client, ...config})`.

Without the overhead of the client - viem suggests that this can improve tree shaking, and recommends it for devs who are "pendantic about bundle size" 🤣 https://viem.sh/docs/clients/custom#tree-shaking